tests/nix/simple/default.nix

```diff
@@ -0,0 +1,21 @@
+let
+  # Pinning nixpkgs to specific release
+  # To get sha256 use "nix-prefetch-git <url> --rev <commit>"
+  commitRev="5574b6a152b1b3ae5f93ba37c4ffd1981f62bf5a";
```

Comment:
should we expose this via `runtime.txt`  like we do for MRAN which gets set via `r-<date>` in `runtime.txt`. I assume everyone who uses nix will want to set this SHA to something so we could move this code from the `default.nix` file? The advantage would be less typing and that this behaviour would be enforced because you have to put the SHA in the `runtime.txt`.

What do you think how that fits with nix user's expectations?

Reply:
I think we should avoid using runtime.txt for anything where a provider has a mechanism to specify something like this. It's exclusively an escape hatch where there's no better way to specify (i.e. only in R, requirements.txt, not needed and shouldn't be supported for Julia, conda, etc.).